chat-plugins/info.js

```diff
@@ -1004,6 +1004,209 @@ var commands = exports.commands = {
+					if (Number.isInteger(Number(searchedWords[k]))) {
+						if (basePower) return this.sendReplyBox("Only specify a number for base power once.");
+						basePower = parseInt(searchedWords[k]);
+						if (basePower < 70 || 100 < basePower) return this.sendReplyBox('No berries inflict ' + basePower + 'bp damage when used with Natural Gift.');
+					}
+				}
+			}
+
+			for (var n in Tools.data.Items) {
+				var item = Tools.getItem(n);
+				if (!item.isBerry) continue;
+
+				if (basePower && type) {
+					if (item.naturalGift.basePower === basePower && item.naturalGift.type === type) foundItems.push(item.name);
+				} else if (basePower) {
+					if (item.naturalGift.basePower === basePower) foundItems.push(item.name);
+				} else {
+					if (item.naturalGift.type === type) foundItems.push(item.name);
+				}
+			}
+		} else {
+			var bestMatched = 0;
+			for (var n in Tools.data.Items) {
+				var item = Tools.getItem(n);
+				var matched = 0;
+				var descWords = item.desc.toLowerCase().replace('-', ' ').replace(/[^a-z0-9\s\/]/g, '').replace(/(\D)\./, function (p0, p1) { return p1; }).split(" ");
+
+				for (var k = 0; k < searchedWords.length; k++) {
+					if (descWords.indexOf(searchedWords[k]) >= 0) {
+						matched++;
+					}
+				}
+
+				if (matched >= bestMatched && matched >= (searchedWords.length * 3 / 5)) foundItems.push(item.name);
+				if (matched > bestMatched) bestMatched = matched;
+			}
+
+			// iterate over found items again to make sure they all are the best match
+			for (var l = 0; l < foundItems.length; l++) {
+				var item = Tools.getItem(foundItems[l]);
+				var matched = 0;
+				var descWords = item.desc.toLowerCase().replace('-', ' ').replace(/[^a-z0-9\s\/]/g, '').replace(/(\D)\./, function (p0, p1) { return p1; }).split(" ");
+
+				for (var k = 0; k < searchedWords.length; k++) {
+					if (descWords.indexOf(searchedWords[k]) >= 0) {
+						matched++;
+					}
+				}
+
+				if (matched !== bestMatched) {
+					foundItems.splice(l, 1);
+					l--;
+				}
+			}
+		}
+
+		var resultsStr = "";
+		if (foundItems.length > 0) {
+			if (showAll || foundItems.length <= output + 5) {
+				foundItems.sort();
+				resultsStr += foundItems.join(", ");
+			} else {
+				resultsStr += foundItems.slice(0, output).join(", ") + ", and " + string(foundItems.length - output) + " more. <font color=#999999>Redo the search with 'all' as a search parameter to show all results.</font>";
+			}
+		} else {
+			resultsStr = "No items found. Try a more general search";
+		}
+		return this.sendReplyBox(resultsStr);
+	},
+	itemsearchhelp: ["/itemsearch [move description] - finds items that match the given key words.",
+	"/itemsearch [fling] [base power] [effect] - finds items with the given fling behavior.",
+	"/itemsearch [natural gift] [base power] [type] - finds berries with the given natural gift behavior."],
```

Comment:
I don't really know how to document the help for this at all. It's UI is so loose might be best to not use parameter notation at all. I mean like, you can write `/itemsearch hey, what's that berry that deals an 80bp damage fire move when used with natural gift?` and it will still work. I could make Fling and Natural Gift use more strict syntax, but then it might feel inconsistent with how the general description search works.

I guess the best way to do it would be to take your suggestion of `/itemsearch fling [base power] [effect]` and then add another line that says "`Order of words doesn't matter.`"